python/ray/serve/_private/replica.py

```diff
@@ -845,7 +845,9 @@ async def prepare_for_shutdown(self):
         # We set the del method to noop after successfully calling it so the
         # destructor is called only once.
         async with self.delete_lock:
-            self.metrics_pusher = None
+            if self.metrics_pusher:
+                self.metrics_pusher.__del__()
+                self.metrics_pusher = None
```

Comment:
we should do the same thing as below and overwrite the `__del__` method with a no-op so it isn't called twice

Reply:
It's probably better to move the shutdown logic into an idempotent "shutdown" method, then in `__del__` you can call that method (which will do nothing if shutdown was already called)

Reply:
Great idea! Let me add that to metrics pusher